Assets/Wizards Code/MaxArchxd/Scripts/DangerousAreaScanner.cs

```diff
@@ -0,0 +1,141 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+using WizardsCode.BackgroundAI;
+
+[RequireComponent(typeof(NavMeshAgent))]
+public class DangerousAreaScanner : AbstractSingleton<DangerousAreaScanner>
```

Comment:
I feel that this would be better as a Sense (See LineOfSight as example) rather than a monobehaviour. Though this needs a little more thought before being sure about that.

Reply:
Also, should this really be a singleton? Doesn't that mean that it is impossible to have different collections of dangerous obstacles for different characters?